### PR TITLE
Use BUILD_TAG to tag cloud resources

### PR DESCRIPTION
**Pull request by mtojek**
Close: https://github.com/elastic/elastic-package/issues/754

This PR introduces a new environment variable to tag created cloud resources with Terraform service deployer.

### CODE REVIEW

**Comment by mtojek** [comment]
Instance created correctly, but the "tear down" didn't work. I am investigating this.

Zrzut ekranu 2022-03-24 o 19 35 06

**Comment by mtojek** [comment]
@endorama I figured out why it wasn't removing VMs. The cleanup operation was never called due to an introduced bug. Take a look at this:

```
USER       PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root         1  0.0  0.0   2616   528 pts/0    Ss+  19:13   0:00 /bin/sh -c bash /run.sh
root         8  0.0  0.0   3984  3112 pts/0    S+   19:13   0:00 bash /run.sh
root       187  0.1  0.0   4248  3440 pts/1    Ss   19:13   0:00 /bin/bash
root       346  0.0  0.0   2516   576 pts/0    S+   19:15   0:00 sleep 1
root       347  0.0  0.0   5904  2940 pts/1    R+   19:15   0:00 ps aux
```

Docker can send the SIGNAL to the PID 1, but it doesn't mean that the signal will be passed down, so to properly pass it  we have to use `ENTRYPOINT exec /run`.sh. Then, it's fine:

```
USER       PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root         1  0.2  0.0   3984  3044 pts/0    Ss+  19:50   0:00 bash /run.sh
root        27  0.2  0.0   4248  3432 pts/1    Ss   19:50   0:00 /bin/bash
root       246 47.7  0.2 767008 73856 pts/0    Sl+  19:51   0:01 terraform destroy -auto-approve
root       319  0.0  0.1 770760 53132 pts/0    Sl+  19:51   0:00 .terraform/providers/registry.terraform.io/hashicorp/google/4.5.0/linux_amd64/terraform-provider-google_v4.5.0_x5
root       336  0.0  0.0   5904  2944 pts/1    R+   19:51   0:00 ps aux
```

As you can see, the "destroy" was invoked.

File: test/packages/parallel/gcp/data_stream/compute/_dev/deploy/tf/compute.tf [hunk 1]
```diff
@@ -13,8 +13,8 @@ resource "google_compute_instance" "default" {
   zone         = var.zone
 
   labels = {
-    team   = "integrations"
     run_id = var.TEST_RUN_ID
+    build_tag = var.BUILD_TAG
```
[the remark this answers is not in the document]
Comment (mtojek) [reply test/packages/parallel/gcp/data_stream/compute/_dev/deploy/tf/compute.tf:17]: I can work on this.

[the remark this answers is not in the document]
Comment (mtojek) [reply test/packages/parallel/gcp/data_stream/compute/_dev/deploy/tf/compute.tf]: Fixed

**Review by jsoriano — APPROVED** [review]

File: internal/install/_static/terraform_deployer.yml [hunk 2]
```diff
@@ -3,7 +3,11 @@ services:
   terraform:
     build: .
     tty: true
+    stop_grace_period: 5m
```
[the remark this answers is not in the document]
Comment (mtojek) [reply internal/install/_static/terraform_deployer.yml:6]: It's a bugfix to fix the regression introduced some time ago. Machines weren't destroyed properly. I described it [here](https://github.com/elastic/elastic-package/pull/755#issuecomment-1078080542).

**Comment by endorama** [comment]
I want to implement this tagging in related PRs for `gcp` integration, where do I find some documentation about these available variables?

From my understanding there are 3 variables that can be used:
- `REPO_NAME`
- `CHANGE_ID`
- `BUILD_NUMBER`

I noticed these variables are not set in the `Jenkinsfile`, where are defined? Should they be defined or are Jenkins provided values?

**Comment by mtojek** [comment]
Yes, all of them are provided with Jenkins, see [env-info.json](https://beats-ci.elastic.co/job/Ingest-manager/job/elastic-package/job/PR-755/1/artifact/env-info.json).

**Pull request merged**